geonode/base/forms.py

```diff
@@ -337,16 +337,14 @@ def _define_choicefield(self, item, required, tname, lang):
 
     @staticmethod
     def _get_thesauro_keyword_label(item, lang):
-        qs_local = []
-        qs_non_local = [("", "------")]
-        for key in ThesaurusKeyword.objects.filter(thesaurus_id=item.id):
-            label = ThesaurusKeywordLabel.objects.filter(keyword=key).filter(lang=lang)
-            if label.exists():
-                qs_local.append((label.get().keyword.id, label.get().label))
-            else:
-                qs_non_local.append((key.id, key.alt_label))
+        keyword_id_for_given_thesaurus = ThesaurusKeyword.objects.filter(thesaurus_id=item)
+        qs_keyword_ids = ThesaurusKeywordLabel.objects.filter(lang=lang, keyword_id__in=keyword_id_for_given_thesaurus).values("keyword_id")
+        not_qs_ids = ThesaurusKeywordLabel.objects.exclude(keyword_id__in=qs_keyword_ids).order_by("keyword_id").distinct("keyword_id").values("keyword_id")
+
+        qs_local = list(ThesaurusKeywordLabel.objects.filter(lang=lang, keyword_id__in=keyword_id_for_given_thesaurus).values_list("keyword_id", "label"))
+        qs_non_local = list(keyword_id_for_given_thesaurus.filter(id__in=not_qs_ids).values_list("id", "alt_label"))
 
-        return qs_non_local + qs_local
+        return qs_local + [("", "-------")] + qs_non_local
```

Comment:
Hi @mwallschlaeger I guess here is better to maint the `[("", "-------")]` as first, so during the preview in the select, is at the beginning. Otherwise, I guess it will be in the middle of the list.

Reply:
yes and no. I think its good to have the [("", "-------")] at first. But I would suggest to put it in between the qs_local and the qs_non_local as well, so the user knows that there is a reason why there are appearing entries in another language. 

Again for my usecase with the AGROVOC there are a lot of entries not available in all languages. That means that even if a users searches for something there are always many many entries which are not related to the search.

Imagining looking for something like "v" in a list of car manufactures. Therefore I would suggest an output like:
```
---------
Vandenbrink
Vauxhall
Vector
Vencer
Venturi
Venucia
VinFast
VLF
Volkswagen
Volvo
---------
北京汽车制造厂有限公司
绅宝汽车
银翔
金杯
```

So in this case the second "---------" separates the translated entries from the not translatable ones.

Reply:
Yep having it in both places is a good option